### PR TITLE
chore: upgrade native libraries

### PR DESCRIPTION
**Pull request by charliecruzan-stripe**
## Summary
supersedes https://github.com/stripe/stripe-react-native/pull/1375/

upgrade android to 20.24+
upgrade ios to 23.7.+

### CODE REVIEW

File: stripe-react-native.podspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'json'
 
 package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
-stripe_version = '23.6.0'
+stripe_version = '~> 23.7.0'
```
Comment (ccen-stripe) [inline stripe-react-native.podspec:4]: any reason why we don't use `23.7.0` ?

**Review by ccen-stripe — CHANGES_REQUESTED** [review]

File: stripe-react-native.podspec [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require 'json'
 
 package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
-stripe_version = '23.6.0'
```
Comment (ccen-stripe) [inline stripe-react-native.podspec:4]: Also, could we add a line here
```
When stripe_version is updated, also need to update stripe_version in https://github.com/stripe/stripe-identity-react-native/blob/main/stripe-identity-react-native.podspec
```

File: android/gradle.properties [hunk 3]
```diff
@@ -1,2 +1,2 @@
 StripeSdk_kotlinVersion=1.8.0
-StripeSdk_stripeVersion=20.23.+
+StripeSdk_stripeVersion=20.24.+
```
Comment (ccen-stripe) [inline android/gradle.properties:2]: Could we add a line here:
```
When StripeSdk_kotlinVersion and StripeSdk_stripeVersion is updated, also need to update StripeIdentityReactNative_kotlinVersion and StripeIdentityReactNative_stripeVersion in https://github.com/stripe/stripe-identity-react-native/blob/main/android/gradle.properties
```

File: stripe-react-native.podspec (hunk 1 above)
Comment (charliecruzan-stripe) [reply stripe-react-native.podspec:4]: this matches up to but not including 23.8.0, so we'll get in any patch fixes that `stripe-ios` releases. it's the equivalent of the android's `20.24.+`

Comment (ccen-stripe) [reply stripe-react-native.podspec:4]: ok, will apply the similar change in identity

**Review by ccen-stripe — CHANGES_REQUESTED** [review]

**Pull request merged**